scripts/generate-config-data.js

```diff
@@ -0,0 +1,245 @@
+        }
+      } else {
+        console.warn('Warning: Config option %s.%s is not linked to any CLI option.', section.name, prop.name);
+      }
+    }
+  }
+
+  return data;
+}
+
+function augment (data, extra) {
+  const dataAugmented = Object.assign({}, data);
+  const overwritten = [];
+
+  for (const section in extra) {
+    if (!(section in dataAugmented)) {
+      dataAugmented[section] = extra[section];
+      overwritten.push(section);
+      continue;
+    }
+
+    for (const prop in extra[section]) {
+      if (prop === 'section' || prop === 'description' || prop === 'condition') {
+        dataAugmented[section][prop] = extra[section][prop];
+        continue;
+      }
+
+      overwritten.push(`${section}.${prop}`);
+      dataAugmented[section][prop] = {...dataAugmented[section][prop], ...extra[section][prop]};
+    }
+  }
+
+  // Quick hack to retain the key order of data.extra.json
+  // (make new object and insert data.extra.json props first)
+
+  const dataAugmentedOrdered = {};
+
+  overwritten.forEach(id => {
+    if (!id.includes('.')) {
+      dataAugmentedOrdered[id] = dataAugmented[id];
+    } else {
+      const [section, prop] = id.split('.');
+      dataAugmentedOrdered[section] = dataAugmentedOrdered[section] || {};
+      dataAugmentedOrdered[section][prop] = dataAugmented[section][prop];
+    }
+  });
+
+  for (const section in dataAugmented) {
+    if (overwritten.includes(section)) {
+      continue;
+    }
+
+    dataAugmentedOrdered[section] = dataAugmentedOrdered[section] || {};
+
+    for (const prop in dataAugmented[section]) {
+      if (overwritten.includes(`${section}.${prop}`)) {
+        continue;
+      }
+
+      dataAugmentedOrdered[section][prop] = dataAugmented[section][prop];
+    }
+  }
+
+  return dataAugmentedOrdered;
+}
+
+(async function () {
+  const source = await fetchSource();
+
+  // Parse CLI options
+
+  const cliOptions = getCliOptions(source);
+  const configCliOptions = makeCliConfigTree(cliOptions);
+
+  // Parse config structs
+
+  const configSections = getStructFields('Config', source).map(({name: section, type: sectionStructName}) =>
+      ({
+        name: section,
+        name_friendly: sectionStructName,
+        props: getStructFields(sectionStructName, source).map(prop => {
+          prop.type = prop.type.toLowerCase().replace(/vec<(.+)>/, '$1[]').replace(/u16|u32|u64|usize/, 'number');
+          return prop;
+        })
+      })
+  );
+
+  // Hydrate config structs with CLI options
+
+  const data = hydrateConfigWithCli(configSections, configCliOptions);
+
+  // Augment with data.extra.json
+
+  const extra = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../src/data.extra.json'), 'UTF-8'));
+  const dataAugmented = augment(data, extra);
+
+  // Write to file
+
+  fs.writeFileSync(path.resolve(__dirname, '../src/data.compiled.json'), JSON.stringify(dataAugmented, null, 2));
+})();
```

Comment:
Should handle errors from the returned promise (by `console.error` and `process.exit(1)`)